File: Python Version/Deck.py
from random import shuffle
from typing import List
from Card import Card


class Deck:
    """Deck class"""
    def __init__(self) -> None:
        self.cards: List[Card] = []
        for i in range(2, 15):
            for j in range(4):
                self.cards.append(Card(i, j))
        for i in range(7):
            shuffle(self.cards)
        
    def rm_card(self) -> Card:
        if len(self.cards) == 0:
            return
        
        return self.cards.pop()
    

File: Java version/Deck.java
import java.util.ArrayList;
import java.util.Collections;
// import java.util.List;
import Card.java;


public class Deck {
    private ArrayList<Card> cards;

    public Deck() {
        this.cards = new ArrayList<>();
        for (int i = 2; i <= 14; i++) {
            for (int j = 0; j < 4; j++) {
                this.cards.add(new Card(i, j));
            }
        }
        for (int i = 0; i < 7; i++) {
            Collections.shuffle(this.cards);
        }
    }

    public Card removeCard() {
        if (this.cards.size() == 0) {
            return null;
        }
        return this.cards.remove(this.cards.size() - 1);
    }

    public ArrayList<Card> getCards() {
        return this.cards;
    }
}
